kanopi/kanopi-pack-react
# Watch Mode

## Description
> As a developer, I need to add a Watch mode to Kanopi Pack React to support systems without the ability to use the Development Server, easily or at all.

- Add a watch mode to standard command set, uses the watch polling options for Polling mode
 
## Acceptance Criteria
* Existing Development and Production modes work
* Watch mode successfully loads, like Production mode

## Steps to Validate
1. Use the feature branch for KP Standard and KP React. To update your projects dependencies, add them as `npm i https://github.com/kanopi/kanopi-pack.git#feature/watch` and `npm i https://github.com/kanopi/kanopi-pack-react.git#feature/watch`
2. Add a `scripts` key in your projects `package.json` to reference the watch command, i.e.: `"watch": "kanopi-pack-react react watch"`
3. Run the watch command: `npm run watch`, verify it builds the bundle and continues to listen without crashing.
4. Load a page on the site using the assets 
5. Change an asset file, CSS, JS, etc and verify the command line shows the bundle rebuild (it will take about the same time production takes to build)
6. Refresh the page and verify the changes show
7. Close watch mode with Ctrl+C
